packages/compiler/src/typekit/kits/builtin.ts

```diff
@@ -142,6 +148,36 @@ declare module "../define-kit.js" {
 
 defineKit<TypekitExtension>({
   builtin: {
+    is(type: Type): boolean {
+      // Model property does not have a namespace,
+      // so we will use the Model where it is defined.
+      if (type.kind === "ModelProperty" && type.model) {
+        type = type.model;
+      }
+
+      // Consider the type not built-in if it has no namespace
+      // TypeKit might create types without a namespace
+      const ns = (type as any).namespace as Namespace | undefined;
+      if (!ns) {
+        return false;
+      }
+
+      const globalNs = this.program.getGlobalNamespaceType();
+      // If it’s already in the global root, it's not in global.TypeSpec
+      if (ns === globalNs) {
+        return false;
+      }
+
+      // Find the immediate child of `globalNs` in this namespace's ancestry.
+      const topLevel = getImmediateChildOfGlobal(ns, globalNs);
+      if (!topLevel) {
+        return false;
+      }
+
+      // Finally, compare by identity
+      const typeSpecNs = globalNs.namespaces.get("TypeSpec");
+      return topLevel === typeSpecNs;
```

Comment:
is that always working when using mutators?

Reply:
I seem to recall at some point that we had special-cased TypeSpec so as to avoid mutating built-in types. Is that not the case? If it's not, then I think we might have more bugs in other places, e.g. the intrinsic is checks use reference equality.